### DISCUSSION TITLE
Problem confirming new registrations

### DISCUSSION THREAD

**Discussion by jgdealmeida** [dani-garcia/vaultwarden#3952] [Q&A]
### Subject of the issue

Users who are registering on vaultwarden are taking a long time to appear in the "Needs confirmation" tab and when they do, when confirming the registration through the web interface, I encounter the following error Cannot read properties of null (reading 'buffer') and action is not completed.

### Deployment environment
* Install method: Docker image

* vaultwarden version: 1.29.2

* Install method: docker

* Clients used: irrelevant

* Reverse proxy and version: irrelevant

* MySQL/MariaDB or PostgreSQL version: MariaDB 10.5.12

* Other relevant details: All recommended installation settings are being applied. The service has now been updated and restarted.

![vaultwarden](https://github.com/dani-garcia/vaultwarden/assets/58986411/8143172f-17e9-486e-9afd-623b3961ca71)

**Comment by stefan0xC** [comment]
Can you post the support string from the diagnostics page in the `/admin` panel? Because "All recommended installation settings are being applied" is not really telling us anything about your setup. And you might also want to add some logs from the server when you try to confirm a user for additional context.

Also the finger-print phrase is missing from the confirm dialog. Not sure why. Can you check and make sure that this user has a `password_hash` set in the database? (It would probably not work if the user has been accepted without creating an account first, which was for example possible with an LDAP import before d892880dd2f96dd9f72a6ce9437bf638719a4bcb has been added in the latest release.)

**Comment by jgdealmeida** [comment]
Hello. I was using the recommendations for Nginx from this article: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

But I will try to carry out the procedures you indicated at the end of the message.

As soon as I do the tests I will give feedback again.

**Comment by jgdealmeida** [comment]
### Suport string

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.29.2
* Web-vault version: v2023.7.1
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: MySQL
* Database version: 10.5.8-MariaDB-1:10.5.8+maria~bionic
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

json
{
"_duo_akey": null,
"_enable_duo": false,
"_enable_email_2fa": false,
"_enable_smtp": true,
"_enable_yubico": true,
"_icon_service_csp": "",
"_icon_service_url": "",
"_ip_header_enabled": true,
"_smtp_img_src": "cid:",
"admin_ratelimit_max_burst": 3,
"admin_ratelimit_seconds": 300,
"admin_session_lifetime": 20,
"admin_token": "***",
"allowed_iframe_ancestors": "",
"attachments_folder": "data/attachments",
"auth_request_purge_schedule": "30 * * * * *",
"authenticator_disable_time_drift": false,
"data_folder": "data",
"database_conn_init": "",
"database_max_conns": 10,
"database_timeout": 30,
"database_url": "***://*************************************************************",
"db_connection_retries": 15,
"disable_2fa_remember": false,
"disable_admin_token": false,
"disable_icon_download": false,
"domain": "***://***********************",
"domain_origin": "***://***********************",
"domain_path": "",
"domain_set": true,
"duo_host": null,
"duo_ikey": null,
"duo_skey": null,
"email_attempts_limit": 3,
"email_expiration_time": 600,
"email_token_size": 6,
"emergency_access_allowed": true,
"emergency_notification_reminder_schedule": "0 3 * * * *",
"emergency_request_timeout_schedule": "0 7 * * * *",
"enable_db_wal": true,
"event_cleanup_schedule": "0 10 0 * * *",
"events_days_retain": null,
"extended_logging": true,
"helo_name": null,
"hibp_api_key": null,
"icon_blacklist_non_global_ips": false,
"icon_blacklist_regex": null,
"icon_cache_folder": "data/icon_cache",
"icon_cache_negttl": 259200,
"icon_cache_ttl": 2592000,
"icon_download_timeout": 10,
"icon_redirect_code": 302,
"icon_service": "internal",
"incomplete_2fa_schedule": "30 * * * * *",
"incomplete_2fa_time_limit": 3,
"invitation_expiration_hours": 120,
"invitation_org_name": "Quality TI",
"invitations_allowed": true,
"ip_header": "X-Real-IP",
"job_poll_interval_ms": 30000,
"log_file": null,
"log_level": "Info",
"log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
"login_ratelimit_max_burst": 10,
"login_ratelimit_seconds": 60,
"org_attachment_limit": null,
"org_creation_users": "",
"org_events_enabled": false,
"org_groups_enabled": false,
"password_hints_allowed": true,
"password_iterations": 100000,
"push_enabled": false,
"push_installation_id": "***",
"push_installation_key": "***",
"push_relay_uri": "https://push.bitwarden.com",
"reload_templates": false,
"require_device_email": false,
"rsa_key_filename": "data/rsa_key",
"send_purge_schedule": "0 5 * * * *",
"sendmail_command": null,
"sends_allowed": true,
"sends_folder": "data/sends",
"show_password_hint": false,
"signups_allowed": true,
"signups_domains_whitelist": "",
"signups_verify": false,
"signups_verify_resend_limit": 6,
"signups_verify_resend_time": 3600,
"smtp_accept_invalid_certs": false,
"smtp_accept_invalid_hostnames": false,
"smtp_auth_mechanism": null,
"smtp_debug": false,
"smtp_embed_images": true,
"smtp_explicit_tls": false,
"smtp_from": "**************************",
"smtp_from_name": "Adm Sistemas",
"smtp_host": "******************",
"smtp_password": "***",
"smtp_port": 587,
"smtp_security": "starttls",
"smtp_ssl": true,
"smtp_timeout": 15,
"smtp_username": "**************************",
"templates_folder": "data/templates",
"tmp_folder": "data/tmp",
"trash_auto_delete_days": null,
"trash_purge_schedule": "0 5 0 * * *",
"use_sendmail": false,
"use_syslog": false,
"user_attachment_limit": null,
"web_vault_enabled": true,
"web_vault_folder": "web-vault/",
"websocket_address": "0.0.0.0",
"websocket_enabled": true,
"websocket_port": 3012,
"yubico_client_id": null,
"yubico_secret_key": null,
"yubico_server": null
}
</details>



### logs from server
![Clipboard - 3 de Outubro de 2023 às 14_24](https://github.com/dani-garcia/vaultwarden/assets/58986411/5053e5f5-6b04-4f67-89fd-33edfb63ba86)

And the users in question are without the `password_hash`

**Reply by stefan0xC** [reply]
> And the users in question are without the `password_hash`

That means that they have not registered as a user yet and can't be confirmed. I'm not sure if this is due to an already fixed bug or caused by something else. 

Did you use LDAP import or something like that? Did you always have mail enabled?

**Reply by jgdealmeida** [reply]
That's it, this and other users have already registered and can even use the tool and store their credentials.

We do not use LDAP and always use email.

**Comment by BlackDex** [comment]
I'm not seeing any issues in the logs.
Maybe it's the reverse proxy which blocks stuff.

**Reply by jgdealmeida** [reply]
Absolutely no errors appear in the logs when I try to confirm the registration.

**Reply by BlackDex** [reply]
Then, check what Vaultwarden returns. There seems to be a value returning `null` or an empty string maybe, which causes the web-vault to error.
